src/main/kotlin/org/jitsi/rtp/rtcp/rtcpfb/payload_specific_fb/RtcpFbRembPacket.kt

```diff
@@ -0,0 +1,189 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.rtp.rtcp.rtcpfb.payload_specific_fb
+
+import org.jitsi.rtp.extensions.bytearray.cloneFromPool
+import org.jitsi.rtp.extensions.bytearray.putInt
+import org.jitsi.rtp.rtcp.RtcpHeaderBuilder
+import org.jitsi.rtp.rtcp.rtcpfb.RtcpFbPacket
+import org.jitsi.rtp.rtcp.rtcpfb.payload_specific_fb.RtcpFbRembPacket.Companion.BR_LEN
+import org.jitsi.rtp.rtcp.rtcpfb.payload_specific_fb.RtcpFbRembPacket.Companion.BR_OFF
+import org.jitsi.rtp.rtcp.rtcpfb.payload_specific_fb.RtcpFbRembPacket.Companion.NUM_SSRC_LEN
+import org.jitsi.rtp.rtcp.rtcpfb.payload_specific_fb.RtcpFbRembPacket.Companion.NUM_SSRC_OFF
+import org.jitsi.rtp.rtcp.rtcpfb.payload_specific_fb.RtcpFbRembPacket.Companion.REMB_LEN
+import org.jitsi.rtp.rtcp.rtcpfb.payload_specific_fb.RtcpFbRembPacket.Companion.REMB_OFF
+import org.jitsi.rtp.rtcp.rtcpfb.payload_specific_fb.RtcpFbRembPacket.Companion.SSRCS_OFF
+import org.jitsi.rtp.util.BufferPool
+import org.jitsi.rtp.util.RtpUtils
+import org.jitsi.rtp.util.getBitsAsInt
+import org.jitsi.rtp.util.getByteAsInt
+import org.jitsi.rtp.util.getIntAsLong
+import org.jitsi.rtp.util.getShortAsInt
+
+/**
+ * https://tools.ietf.org/html/draft-alvestrand-rmcat-remb-03
+ *
+ * 0                   1                   2                   3
+ * 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |V=2|P| FMT=15  |   PT=206      |             length            |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |                  SSRC of packet sender                        |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |                  SSRC of media source                         |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |  Unique identifier 'R' 'E' 'M' 'B'                            |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |  Num SSRC     | BR Exp    |  BR Mantissa                      |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |   SSRC feedback                                               |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |  ...                                                          |
+ *
+ * @author George Politis
+ * @author Boris Grozev
+ */
+class RtcpFbRembPacket(
+    buffer: ByteArray,
+    offset: Int,
+    length: Int
+) : PayloadSpecificRtcpFbPacket(buffer, offset, length) {
+
+    /**
+     * The exponential scaling of the mantissa for the maximum total media
+     * bit rate value, ignoring all packet overhead.
+     */
+    val exp: Int
+        get() = getBrExp(buffer, offset)
```

Comment:
we've made the packet types immutable, so should be able to change this to calculate it once rather than every time:

`val exp: Int = getBrExp(buffer, offset)`

Reply:
(same for the ones below)

Reply:
(could also make it lazy instead, if it's expensive and we may not always access it)